File: lib/Models/Patient.dart


class Patient {
  String patientId;
  String fullName;
  String address;
  String allergies;
  String bloodType;
  String chronicDiseases;
  DateTime createAt;
  DateTime dateOfBrith;
  String emergencyContactName;
  String emergencyContactPhone;
  String gender;
  double height;
  DateTime updateAt;

  Patient({
    required this.patientId,
    required this.fullName,
    required this.address,
    required this.allergies,
    required this.bloodType,
    required this.chronicDiseases,
    required this.createAt,
    required this.dateOfBrith,
    required this.emergencyContactName,
    required this.emergencyContactPhone,
    required this.gender,
    required this.height,
    required this.updateAt,
  });

  Map<String, dynamic> toMap() {
    return {
      'patientId': patientId,
      'fullName': fullName,
      'address': address,
      'allergies': allergies,
      'bloodType': bloodType,
      'chronicDiseases': chronicDiseases,
      'createAt': createAt,
      'dateOfBrith': dateOfBrith,
      'emergencyContactName': emergencyContactName,
      'emergencyContactPhone': emergencyContactPhone,
      'gender': gender,
      'height': height,
      'updateAt': updateAt,
    };

  }

  factory Patient.fromMap(Map<String, dynamic> map) {
    return Patient(
      patientId: map['patientId'],
      fullName: map['fullName'],
      address: map['address'],
      allergies: map['allergies'],
      bloodType: map['bloodType'],
      chronicDiseases: map['chronicDiseases'],
      createAt: map['createAt'] as DateTime,
      dateOfBrith: map['dateOfBrith'] as DateTime,
      emergencyContactName: map['emergencyContactName'],
      emergencyContactPhone: map['emergencyContactPhone'],
      gender: map['gender'],
      height: map['height'],
      updateAt: map['updateAt'] as DateTime,
    );
  }

}

File: Connection_API_Cor_ASPNET/ConnectionStringAPI/ConnectionStringAPI/ResultDB/Patient.cs
using System;
using System.Collections.Generic;

namespace ConnectionStringAPI.ResultDB;

public partial class Patient
{
    public string PatientId { get; set; }

    public string FullName { get; set; } = null!;

    public string? Gender { get; set; }

    public DateOnly? DateOfBirth { get; set; }

    public string? Address { get; set; }

    public string? Phone { get; set; }

    public string? BloodType { get; set; }

    public decimal? Height { get; set; }

    public decimal? Weight { get; set; }

    public string? Allergies { get; set; }

    public string? ChronicDiseases { get; set; }

    public DateTime? CreatedAt { get; set; }

    public DateTime? UpdateAt { get; set; }

    public string? UserId { get; set; }

    public virtual ICollection<Appointment> Appointments { get; set; } = new List<Appointment>();

    public virtual ICollection<Chatwithdoctor> Chatwithdoctors { get; set; } = new List<Chatwithdoctor>();

    public virtual ICollection<Payment> Payments { get; set; } = new List<Payment>();

    public virtual User? User { get; set; }
}
